### PR TITLE
Injector stats

### LINKED ISSUE #158: Stats for POs
Produce stats for every page object that's being injected during the run. Stats should look like:

```
scrapy-poet/page-objects/my_project.page.AmazonProductPage 56
scrapy-poet/page-objects/my_project.page.AmazonListingPage 15
```

### PR DESCRIPTION
**Pull request by proway2**
Fixes #158 

- For every internal dependency that's being injected there will be a record in Scrapy stats:
```
poet/injector/tests.test_injection.PricePO 15
```
- Other places in the injector, where stats are produced, have updated prefix `poet/*`
- `pytest-twisted` is added to `requirements-dev.txt` otherwise it's impossible to set up the local env (tox works fine in any case)

### CODE REVIEW

**Review by Gallaecio — APPROVED** [review]
Looks good to me.

I do wonder if we should mention in the Stats page of the documentation the stats that are logged beyond those that are created using web-poet stats. Or maybe just mention the kind of stuff that’s logged, without going into detail.

**Comment by proway2** [comment]
@Gallaecio I'll add a short note to docs, without details though.

**Comment by proway2** [comment]
@Gallaecio when this could potentially be merged and released?

File: tests/test_injection.py [hunk 1]
```diff
@@ -305,6 +305,82 @@ def to_item(self):
         return item
 
 
+@attr.s(auto_attribs=True)
+class TestItem:
+    foo: int
+    bar: str
+
+
+class TestItemPage(ItemPage, Returns[TestItem]):
```
Comment (kmike) [inline tests/test_injection.py:314]:
```suggestion
class TestItemPage(ItemPage[TestItem]):
```

Comment (proway2) [reply tests/test_injection.py:314]: Fixed.

File: scrapy_poet/injection.py [hunk 2]
```diff
@@ -161,6 +162,8 @@ def build_instances(self, request: Request, response: Response, plan: andi.Plan)
         for cls, kwargs_spec in plan.dependencies:
             if cls not in instances.keys():
                 instances[cls] = cls(**kwargs_spec.kwargs(instances))
+                cls_fqn = get_fq_class_name(cast(type, cls))
+                self.crawler.stats.inc_value(f"poet/injector/{cls_fqn}")
```
[the remark this answers is not in the document]
Comment (proway2) [reply scrapy_poet/injection.py:166]: Well, I can add logging to the other part too. For this case https://github.com/scrapinghub/scrapy-poet/pull/161/files#diff-11a6fa3f7d3b33fc5f88c86547820b9b29ec5cd54e9913373907c63f620d8321R369-R381 it looks like this now: 

```
poet/injector/tests.test_injection.TestItemPage 1
```

But if we count everything it'll look like this:

```
poet/injector/tests.test_injection.TestItemPage 1
poet/injector/tests.test_injection.TestItem 1
```

Seems a bit misleading to me. I don't have a strong opinion though.

[the remark this answers is not in the document]
Comment (kmike) [reply scrapy_poet/injection.py:166]: Let's consider it a separate feature, if we decide to implement it.

**Comment by kmike** [comment]
Thanks @proway2!

**Pull request merged**